x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackSettings.java

```diff
@@ -41,7 +41,7 @@ private XPackSettings() {
     public static final Setting<Boolean> CCR_ENABLED_SETTING = Setting.boolSetting("xpack.ccr.enabled", true, Property.NodeScope);
 
     /** Setting for enabling or disabling data frame. Defaults to true. */
-    public static final Setting<Boolean> DATA_FRAME_ENABLED = Setting.boolSetting("xpack.data_frame.enabled", true,
+    public static final Setting<Boolean> TRANSFORM_ENABLED = Setting.boolSetting("xpack.transform.enabled", true,
             Setting.Property.NodeScope);
```

Comment:
I know we said we wouldn't provide BWC for this, but it's actually trivial to do it and it might make someone's life easier:

```
    // TODO remove before 8.0.0 release
    private static final Setting<Boolean> DATA_FRAME_ENABLED = Setting.boolSetting("xpack.data_frame.enabled", true,
            Setting.Property.NodeScope, Setting.Property.Deprecated);
    public static final Setting<Boolean> TRANSFORM_ENABLED = Setting.boolSetting("xpack.transform.enabled", DATA_FRAME_ENABLED,
            Setting.Property.NodeScope);
```

The `Setting` class completely takes care of the deprecation warnings.

So maybe we should?

We'll have other tweaks we need to make before 8.0.0 release, such as the role names, so it's not even like this creates much work for us.